app/models/spree/line_item_decorator.rb

```diff
@@ -73,7 +73,7 @@
 
   def variant
     # Overridden so that LineItems always have access to soft-deleted Variant attributes
-    Spree::Variant.unscoped { super }
+    Spree::Variant.unscoped { super } || Spree::Variant.unscoped.find(self.variant_id)
```

Comment:
Nice!
We need a comment explaining this...

Reply:
I wonder what's the impact of this in terms of performance. Does it execute the query every time you do `line_item.variant`? not leveraging the cache?

Reply:
it will only execute the query if super returns nil, in most cases it will not go there.
but that's a really good point, we need to test impact of rail4 on our reports, I added to my notes :+1:
For now, I think we should go ahead with this.

Reply:
I have improved the comment now.

Reply:
@sauloperez if you are ok with this we can merge.